### PR TITLE
Decorating error message for nested configuration

### PR DESCRIPTION
**Pull request by huguesBouvier**
Adding to the error message for nested

### CODE REVIEW

File: aziot/src/internal/common.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ pub async fn resolve_and_tls_handshake(endpoint: hyper::Uri, hostname_display: &
     let _ = hyper_openssl::HttpsConnector::new()
         .with_context(|| {
             anyhow!(
-                "Could not connect to {} : could not create TLS connector",
+                "Could not connect to {} : could not create TLS connector. In nested configuration make sure root certificate and parent root cerificate are identical",
```
Comment (arsing) [inline aziot/src/internal/common.rs:97]:
```
If this is a nested Edge configuration, make sure the root certificate and the parent root certificate are identical.
```

(`cerificate` -> `certificate`, punctuation, grammar.)

---

Also, this message is going to be confusing for someone who a) isn't running IoT Edge in a nested Edge configuration, or b) isn't running IoT Edge at all. Is there a way to know for sure whether the user is running in a nested Edge configuration? Maybe based on the IoT Hub hostname in IS config? ( @daprilik , @massand , @gordonwang0 )

**Review by arsing — CHANGES_REQUESTED** [review]

File: aziot/src/internal/common.rs (hunk 1 above)
Comment (arsing) [reply aziot/src/internal/common.rs:97]: Also also, is this "(parent) root certificate" error supposed to be here, such that it's reported for both DPS and Hub checks? Or is it only supposed to be for hub checks? If the latter, it should be done in the caller of this function in `host_connect_iothub.rs`, not here.

Comment (massand) [reply aziot/src/internal/common.rs:97]: This PR comes a tad early. The parent vs service hostname split has not happened yet in IS, which is required for nested topology awareness. That would be a separate check and this error message should not be here.

Comment (huguesBouvier) [reply aziot/src/internal/common.rs:97]: "Also, this message is going to be confusing for someone who a) isn't running IoT Edge in a nested Edge configuration" I agree with you. However, unless I am wrong,  the only way to know we are in nested configuration is the parent_hostname which is not accessible from here.

Comment (huguesBouvier) [reply aziot/src/internal/common.rs:97]: "If the latter, it should be done in the caller of this function in host_connect_iothub.rs, not here." Agreed

Comment (huguesBouvier) [reply aziot/src/internal/common.rs:97]: "This PR comes a tad early. The parent vs service hostname split has not happened yet in IS, which is required for nested topology awareness. That would be a separate check and this error message should not be here." -> Gotcha! That will be much better. That's what I was originally trying to do. Please keep me posted, I'll remake that PR once I can

**Review by huguesBouvier — COMMENTED** [review]
Closing for now. Will reopen once service vs parenthostname split has happened

**Review by huguesBouvier — COMMENTED** [review]
Re-opening this PR. There is an idea I want to try:
If iotHub name end with .azure-devices.net we know we high certainty the device is not nested.

**Pull request closed without merging**